test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Tests.ps1

```diff
@@ -0,0 +1,194 @@
+        BeforeAll {
+            Push-ModulePath $assetPath
+            $modules = Get-Module -ListAvailable -SkipCompatibilityCheck
+        }
+
+        AfterAll {
+            Pop-ModulePath
+        }
+
+        It "Lists all modules from the module path with -ListAvailable for PSEdition <Edition>" -TestCases ($successCases + $failCases) -Skip:(-not $IsWindows) {
+            param($Edition, $ModuleName)
+
+            $modules.Name | Should -Contain $ModuleName
+        }
+    }
+
+    Context "Loading from checked paths by absolute path with -SkipCompatibilityCheck" {
+        It "Lists compatible modules with -ListAvailable for PSEdition <Edition>" -TestCases ($successCases + $failCases) -Skip:(-not $IsWindows) {
+            param($Edition, $ModuleName)
+
+            $modules = Get-Module -ListAvailable -SkipCompatibilityCheck (Join-Path -Path $assetPath -ChildPath $ModuleName)
+
+            $modules.Name | Should -Contain $ModuleName
+        }
+    }
+}
+
+Describe "Import-Module from CompatiblePSEditions-checked paths" -Tag "CI" {
+    BeforeAll {
+        $successCases = @(
+            @{ Edition = "Core,Desktop"; ModuleName = "BothModule"; Result = $true },
+            @{ Edition = "Core"; ModuleName = "CoreModule"; Result = $true }
+        )
+
+        $failCases = @(
+            @{ Edition = "Desktop"; ModuleName = "DesktopModule"; Result = $true },
+            @{ Edition = "<None>"; ModuleName = "NeitherModule"; Result = $true }
+        )
+
+        $allModules = ($successCases + $failCases).ModuleName
+
+        $assetPath = Join-Path $PSScriptRoot "assets" "CompatiblePSEditionsModules"
+        # Emulate the System32 module path for tests
+        [System.Management.Automation.Internal.InternalTestHooks]::SetTestHook("TestWindowsPowerShellPSHomeLocation", $assetPath)
+    }
+
+    AfterAll {
+        [System.Management.Automation.Internal.InternalTestHooks]::SetTestHook("TestWindowsPowerShellPSHomeLocation", $null)
+    }
+
+    AfterEach {
+        Get-Module $allModules | Remove-Module -Force
+    }
+
+    Context "Imports from module path" {
+        BeforeAll {
+            Push-ModulePath $assetPath
+        }
+
+        AfterAll {
+            Pop-ModulePath
+        }
+
+        It "Successfully imports compatible modules from the module path with PSEdition <Edition>" -TestCases $successCases -Skip:(-not $IsWindows) {
+            param($Edition, $ModuleName, $Result)
+
+            Import-Module $ModuleName
+            [scriptblock]::Create("Test-$ModuleName").Invoke() | Should -Be $Result
+        }
+
+        It "Fails to import incompatible modules from the module path with PSEdition <Edition>" -TestCases $failCases -Skip:(-not $IsWindows) {
+            param($Edition, $ModuleName, $Result)
+
+            [scriptblock]::Create("Import-Module $ModuleName -ErrorAction 'Stop'; Test-$ModuleName") | Should -Throw -ErrorId "Modules_PSEditionNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"
```

Comment:
Why not just do `{Import-Module $ModuleName -ErrorAction Stop;  & "Test-$ModuleName"}  | Should -Throw -ErrorId "Modules_PSEditionNotSupported,Microsoft.PowerShell.Commands.ImportModuleCommand"`